### PR TITLE
Enhance rspconfig sshcfg command (OpenBMC) by using multi-processes

### PR DESCRIPTION
**Pull request by xuweibj**
Befor modify:
```
# time rspconfig fw1738 sshcfg
mid05tor12cn13: ssh keys copied to 172.11.139.13
mid05tor12cn11: ssh keys copied to 172.11.139.11
mid05tor12cn15: ssh keys copied to 172.11.139.15
mid05tor12cn02: ssh keys copied to 172.11.139.2
mid05tor12cn05: ssh keys copied to 172.11.139.5

real	0m24.000s
user	0m0.105s
sys	0m0.000s
```

After modify:
```
# time rspconfig fw1738 sshcfg
mid05tor12cn02: ssh keys copied to 172.11.139.2
mid05tor12cn05: ssh keys copied to 172.11.139.5
mid05tor12cn11: ssh keys copied to 172.11.139.11
mid05tor12cn13: ssh keys copied to 172.11.139.13
mid05tor12cn15: ssh keys copied to 172.11.139.15

real	0m6.933s
user	0m0.100s
sys	0m0.010s
```

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 1]
```diff
@@ -1966,18 +1968,52 @@ sub rspconfig_sshcfg_response {
     my $node = shift;
     my $response = shift;
 
-    my $response_info = decode_json $response->content; 
+    if ($node_info{$node}{cur_status} eq "RSPCONFIG_SSHCFG_REQUEST") {
+        my $child = xCAT::Utils->xfork;
+        if (!defined($child)) {
+            xCAT::SvrUtils::sendmsg("Failed to fork child process to upload firmware image.", $callback, $node);
```
Comment (chenglch) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: Please change this message, here is rspconfig, not rflash

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 2]
```diff
@@ -1298,7 +1297,12 @@ sub deal_with_response {
         return;    
     }
 
-    $status_info{ $node_info{$node}{cur_status} }->{process}->($node, $response); 
+    if ($status_info{ $node_info{$node}{cur_status} }->{process}) {
+        $status_info{ $node_info{$node}{cur_status} }->{process}->($node, $response);
+    } else {
+        xCAT::SvrUtils::sendmsg([1,"Internal error, plase the check the rest handler for current status $node_info{$node}{cur_status}"]);
```
Comment (chenglch) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: This is process handler, not rest api handler

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 3]
```diff
@@ -1966,18 +1968,52 @@ sub rspconfig_sshcfg_response {
...
-        open(FILE, ">$home/.ssh/copy.sh")
-          or die "cannot open file $home/.ssh/copy.sh\n";
-        print FILE "#!/bin/sh
+    my $bmcip = $node_info{$node}{bmc};
+    my $userid = $node_info{$node}{username};
+    my $userpw = $node_info{$node}{password};
+
+    my $home = xCAT::Utils->getHomeDir("root");
+    #generate the copy.sh to do real work on target bmc
+    open(FILE, ">$home/.ssh/copy.sh")
```
Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: since this is run in multiple processes, i am afraid this will be a race condition to write the same file

Comment (immarvin) [reply xCAT-server/lib/xcat/plugins/openbmc.pm]: suggest to generate the copy.sh before forking child processes outside the sshcfg_process , then pass `$userid` as argument to `copy.sh` for each node

**Comment by immarvin** [comment]
agree to merge

**Pull request merged**